kernel/kernel-biosdk-provider/src/main/java/io/mosip/kernel/biosdk/provider/impl/BioProviderImpl_V_0_8.java

```diff
@@ -48,35 +53,72 @@
 
 	@Override
 	public boolean verify(List<BIR> sample, List<BIR> record, BiometricType modality, Map<String, String> flags) {		
-		sample = sample.stream().filter(obj -> modality == BiometricType.fromValue(obj.getBdbInfo().getType().get(0).value()))
-			.collect(Collectors.toList());		
-		return match("AUTH", sample, record.toArray(new BIR[record.size()]), modality, flags);
+		BiometricRecord galleryRecord = getBiometricRecord(record.toArray(new BIR[record.size()]));
```

Comment:
There should be no logic change except the changes related to refactoring. Can we keep the old logics in BioProviderImpl_V_0_8.java , but just perform refactoring on that.